### ISSUE TITLE
Better feedback when publishing/unpublishing

### ISSUE THREAD

**Issue by gashcrumb** [syndesisio/syndesis#2343]
## This is a...

<pre><code>
[x] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

When publishing or unpublishing an integration, the 'Pending' spinner is not enough feedback.  There's a lot of factors that play a role in how long an integration takes to publish, and some of these factors can be out of our control.

I think we should expose more information to the user so that there's more feedback, i.e. 'building', 'starting pod' etc. basically more fine-grained states as the integration transitions from unpublished to published.

Perhaps keep the 'Pending' state but then add a second deployment status field perhaps, I dunno, let's discuss...

@kcbabo @dongniwang FYI...

**Comment by kcbabo** [comment]
Would be great to have a checklist/table that updates as deployment progresses through various states, e.g.

- Assembling
- Building
- Deploying
- Starting

**Comment by gashcrumb** [comment]
Yeah, I think on the integration detail page if it's deploying a checklist would be good.  On the integration list page should the user happen to go there I think probably just showing the current stage of the deployment would be fine in place of Publishing/Unpublishing.

**Comment by chirino** [comment]
A link to the OCP console to the thing we are waiting on would be nice too.  Like pod build logs, DC events page, Pod startup logs etc.

**Comment by TovaCohen** [comment]
This looks good to me, except it is still showing the hover help for "Building" as:
Build started, waiting on it to finish.

For the hover help, I suggest:

Assembling - Creating pod resources needed to build the integration
Building - Getting the integration ready to deploy
Deploying - Waiting for deployment of pod that will run the integration
Starting - Waiting for the integration to be ready to run
Publishing - Waiting for the pod to start running the integration

**Comment by gashcrumb** [comment]
@TovaCohen cool, I'll use that in the implementation, and it'll go into the dictionary file so it'll be easy to change later on.

**Comment by kcbabo** [comment]
Hey - just to confirm, wouldn't this be the correct order?
Assembling - Creating pod resources needed to build the integration
Building - Getting the integration ready to deploy
Deploying - Waiting for deployment of pod that will run the integration
Publishing - Waiting for the pod to start running the integration
Starting - Waiting for the integration to be ready to run

**Comment by gaughan** [comment]
Looks great, especially the log link. UX Approved by PM

**Comment by dongniwang** [comment]
Thanks @gaughan. 

@gashcrumb @kcbabo @dhirajsb @TovaCohen - can we confirm what are the stages? And what is the correct order of these stages? So I can update the design to reflect the latest changes.

**Comment by kcbabo** [comment]
From my POV, it's what I mentioned above : Assembling, Building, Deploying, Publishing, Starting

**Comment by gashcrumb** [comment]
+1 on the designs from me.

**Comment by dhirajsb** [comment]
Opened PR https://github.com/syndesisio/syndesis/pull/2947 to address this in the backend. Need to run the new state monitoring endpoint to get publishing state details by @gashcrumb for UI changes.

**Comment by dongniwang** [comment]
See latest UX design in [UX tracker](https://syndesisio.github.io/syndesis-ux/). Closing this issue. 

If we need to address anything new, please open a new issue.
